cpp/src/parquet/encryption/kms_client.h

```diff
@@ -0,0 +1,84 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "parquet/exception.h"
+#include "parquet/platform.h"
+
+namespace parquet {
+namespace encryption {
+
+// This class wraps the key access token of a KMS server. If your token changes over time,
+// you should keep the reference to the KeyAccessToken object and call Refresh() method
+// every time you have a new token.
+class PARQUET_EXPORT KeyAccessToken {
+ public:
+  KeyAccessToken() = default;
+
+  explicit KeyAccessToken(const std::string value) : value_(value) {}
+
+  void Refresh(const std::string& new_value) { value_ = new_value; }
+
+  const std::string& value() const { return value_; }
+
+  void SetDefaultIfEmpty();
```

Comment:
And the default itself is empty? Is it planned to change to something else, or is this method just irrelevant? :-)

Reply:
This function will set `value_` to `"DEFAULT"` if `value_` is empty.

Reply:
Hmm... is it useful to have a _method_ to do this? Normally, a default value should be used... by default (i.e. if I don't set a value, use the default).

Reply:
After checking again, in both case of wrapping and unwrapping, key access token is always set to "DEFAULT" if empty, so I will remove this method and set it from `KmsConnectionConfig` constructor. Thanks!